### PR TITLE
Add external link text code from dfeshiny

### LINKED ISSUE #102: Add external link function
Pinched from [dfe shiny](https://dfe-analytical-services.github.io/dfeshiny/reference/external_link.html)

External links often open in a new tab - need to warn users of this and protect them against any vulnerabilities that this exposes them to

### PR DESCRIPTION
**Pull request by cjrace**
## Overview of changes

This resolves #102 by adding a new function for handling external links in dashboards. It's copied over from [`dfeshiny::external_link()`](https://dfe-analytical-services.github.io/dfeshiny/reference/external_link.html).

This includes adding a data set to the package and associated extra bits, for the `bad_link_text` table, which is just a single column of bad link text that the function uses to validate against. Hoping this will make it easier if anyone else needs to add data too as there'll be an example of how to do it. The list of link text is the same list we use in the [explore education statistics](https://explore-education-statistics.service.gov.uk/) service, so I've started a [codeowners file](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners) to selfishly put myself as the owner for the link text data, that way if there's any changes I should be notified and then can update the list on EES as well! Or that's the theory, never used a code owners file before, so it's a bit of an experiment. As a default I've put Sarah and I as the owners for the repo.

Once merged here, I will update dfeshiny's function to just be a blank wrapper on `shinyGovstyle::external_link()`, that way anyone using the dfeshiny version will still be able to use their code without changes, but we will only have one version of the package code to maintain in this shinyGovstyle package.

## PR Checklist

- [x] I have updated the documentation (if needed)
- [x] I have added or updated tests for these changes (if applicable)
- [x] I have tested these changes locally using `devtools::check()`

## Reviewer instructions

This went through a few rounds of questionable QA on our side, missing some glaring errors ([initial PR](https://github.com/dfe-analytical-services/dfeshiny/pull/46), [second PR](https://github.com/dfe-analytical-services/dfeshiny/pull/48) and [third PR](https://github.com/dfe-analytical-services/dfeshiny/pull/56)) 😅 so in theory is pretty well tested and reviewed already. Give a general check over to see if it behaves as you'd expect.

Worth also testing if the styling attaches properly, I'm not sure if I should also be attaching the GOV.UK CSS as a dependency here too as well as the custom sr-only CSS?

### CODE REVIEW

**Review by sarahmwong — CHANGES_REQUESTED** [review]
Thanks Cam! Looks really neat and like the individual codeowners rules too to keep track of specific changes :) only a couple of teeny typos I spotted - would it also maybe be useful to add an example to the README too to show how to use?

File: data-raw/bad_link_text.R
Comment (cjrace) [reply data-raw/bad_link_text.R]: Hah, yes, I can't tell you how many times I've made that typo... Fixed now

**Comment by cjrace** [comment]
Happy to add examples to the readme and the run example app, though I'll wait until #118 is merged and I can rebase this branch on top of that, as my over eager code formatting with styler could make the conflicts a bit messy if I edit the run_example file in this PR too without waiting!

File: data-raw/bad_link_text.R
Comment (cjrace) [reply data-raw/bad_link_text.R]: Fixed this

**Review by sarahmwong — APPROVED** [review]

**Pull request merged**
